dev/devicelab/bin/tasks/android_release_builds_exclude_dev_dependencies_test.dart

```diff
@@ -8,72 +8,100 @@ import 'dart:io';
 import 'package:flutter_devicelab/framework/apk_utils.dart';
 import 'package:flutter_devicelab/framework/framework.dart';
 import 'package:flutter_devicelab/framework/task_result.dart';
-import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:flutter_devicelab/framework/utils.dart' as utils;
 import 'package:path/path.dart' as path;
 
 Future<void> main() async {
   await task(() async {
     try {
       await runProjectTest((FlutterProject flutterProject) async {
+        utils.section(
+          'Configure plugins to be marked as dev dependencies in .flutter-plugins-dependencies file',
+        );
+
         // Enable plugins being marked as dev dependncies in the .flutter-plugins-dependencies file.
-        await flutter('config', options: <String>['--explicit-package-dependencies']);
+        await utils.flutter('config', options: <String>['--explicit-package-dependencies']);
 
         // Create dev_dependency plugin to use for test.
         final Directory tempDir = Directory.systemTemp.createTempSync(
           'android_release_builds_exclude_dev_dependencies_test.',
         );
         const String devDependencyPluginOrg = 'com.example.dev_dependency_plugin';
+
+        utils.section('Create plugin dev_dependency_plugin that supports Android');
+
         await FlutterPluginProject.create(
           tempDir,
           'dev_dependency_plugin',
           options: <String>['--platforms=android', '--org=$devDependencyPluginOrg'],
         );
 
+        utils.section('Add dev_dependency_plugin as a dev dependency to the Flutter app project');
+
         // Add devDependencyPlugin as dependency of flutterProject.
         await flutterProject.addPlugin(
-          'dev_dependency_plugin',
+          'dev:dev_dependency_plugin',
           options: <String>['--path', path.join(tempDir.path, 'dev_dependency_plugin')],
         );
 
+        utils.section(
+          'Verify the app includes/excludes dev_dependency_plugin as dependency in each build mode as expected',
+        );
         final List<String> buildModesToTest = <String>['debug', 'profile', 'release'];
         for (final String buildMode in buildModesToTest) {
-          section('APK does contain methods from dev dependency in $buildMode mode');
+          final String gradlew = Platform.isWindows ? 'gradlew.bat' : 'gradlew';
+          final String gradlewExecutable = Platform.isWindows ? '.\\$gradlew' : './$gradlew';
+          final RegExp regExpToMatchDevDependencyPlugin = RegExp(
+            r'--- project :dev_dependency_plugin',
+          );
+          final RegExp regExpToMatchDevDependencyPluginWithTransitiveDependencies = RegExp(
+            r'--- project :dev_dependency_plugin\n(\s)*\+--- org.jetbrains.kotlin.*\s\(\*\)\n(\s)*\\---\sio.flutter:flutter_embedding_' +
+                buildMode,
+          );
```

Comment:
If there's any feedback on the regex I used please let me know! My intention was to make the check for release builds much looser in case the dev dependency is included in some way we don't expect. The other builds should always include the dev dependency plugin that depends on the Flutter embedding and because the default language is Kotlin (I think?), the Kotlin plugin as well.

I can include sample outputs here if helpful for review.

Reply:
Seems reasonable for now.